lib/dangermattic/plugins/common/reporter.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+module Danger
+  # Handles reporting messages in the context of Danger, allowing the generation of
+  # warnings, errors, and simple messages in a Danger run.
+  #
+  # @example Reporting a Warning
+  #   reporter.report(message: "This is a warning message", type: :warning)
+  #
+  # @example Reporting an Error
+  #   reporter.report(message: "This is an error message", type: :error)
+  #
+  # @see Automattic/dangermattic
+  # @tags tool, util, danger
+  #
+  class Reporter < Plugin
+    # Report a message to be posted by Danger as an error (failing the build), a warning or a simple message.
+    #
+    # @param message [String] The message to be reported to Danger.
+    # @param type [Symbol] The type of report. Possible values:
+    #                        - :warning (default): Reports a warning.
+    #                        - :error: Reports an error.
+    #                        - :message: Reports a simple message.
```

Comment:
Nitpick. I felt unbalanced having `message` as the argument label and `:message` as a report type.

I thought about it a bit but couldn't find anything better than `text`. AI suggested `content`, which is not that much better.

I looked at the [Danger source](https://github.com/danger/danger/blob/bd89c4d6f3c7650d766eb804000a10f0b2fb98ab/lib/danger/request_sources/github/github_review_unsupported.rb#L13-L19) and they seem to be fine with the same clash. So maybe I'm overthinking it, as it's often the case.

Reply:
I see what you mean -- I had this problem myself here on another level, with the local var `message` and Danger's function `message()`, which I had then to call using `danger.message()`.

The problem is, I think `message` is probably what's best describing what these things are, given Danger's nomenclature 🤷

Reply:
> I think message is probably what's best describing what these things are

Yeah, same. 🤷‍♂️